model: support ModernBERT embedding model

0xtoward opened the pull request:
## Motivation

Add [ModernBERT](https://huggingface.co/answerdotai/ModernBERT-base) (`answerdotai/ModernBERT-base`) as an encoder-only embedding model.

## Modifications

- `srt/models/modernbert.py`: encoder-only model — token embeddings + encoder (global attention via `RadixAttention(ENCODER_ONLY)`; the alternating local / sliding-window layers use an in-model SDPA band mask) + gated MLP + LAST pooler, with HF weight-name mapping. **No attention-backend changes** — global attention rides the standard RadixAttention path (torch_native / triton / flashinfer), local attention is handled in the model.
- `srt/configs/model_config.py`: register `ModernBertModel` / `ModernBertForMaskedLM` as embedding models.
- `test/manual/prefill_only/test_encoder_embedding_models.py`: add `answerdotai/ModernBERT-base` (flashinfer skipped only for fp32, which it doesn't support for `dtype_q`).
- docs: list ModernBERT under embedding models.

## Accuracy

`test_encoder_embedding_models` matches the HF reference (cosine similarity within `1e-5`) for `torch_native` / `triton` (fp16, fp32) and `flashinfer` (fp16), batch ∈ {1, 2}.

<!-- pr-states:start -->
---
### CI States

Latest PR Test (Base): <!-- slot:pr-test:start -->:white_check_mark: [Run #30422328097](https://github.com/sgl-project/sglang/actions/runs/30422328097)<!-- slot:pr-test:end -->
Latest PR Test (Extra): <!-- slot:pr-test-extra:start -->:x: [Run #30422327975](https://github.com/sgl-project/sglang/actions/runs/30422327975)<!-- slot:pr-test-extra:end -->
<!-- pr-states:end -->

0xtoward commented:
Hi @mickqian , thanks for your review. And now I continually receive the notification that 
<img width="1002" height="267" alt="image" src="https://github.com/user-attachments/assets/46ac036f-3997-43be-946e-9d56db8771ae" />
what should I do to make this PR merged?
Thanks!

sarahliu-cisco reviewed (commented):
One question, do you have information about performance and serving throughput? Especially comparing with others like Triton Inference Server?

0xtoward committed:
model: support ModernBERT embeddings

ModernBERT (answerdotai/ModernBERT-base), encoder-only. Global attention uses
RadixAttention(ENCODER_ONLY); the alternating local / sliding-window layers use
an in-model SDPA band mask, so no attention-backend changes are needed. Runs on
torch_native / triton / flashinfer (flashinfer only skipped for fp32, which it
does not support for dtype_q).

- models/modernbert.py: embeddings + encoder (RadixAttention global, SDPA local)
  + gated MLP + LAST pooler; HF weight-name mapping
- configs/model_config.py: register ModernBert{Model,ForMaskedLM} as embedding
- test: add answerdotai/ModernBERT-base to the encoder embedding test
- docs: list ModernBERT under embedding models

0xtoward commented:
Thanks for the approval @mickqian  . Rebased onto latest main as a single clean commit. I think the red checks are unrelated hardware-runner jobs. @JustinTong0323   could you approve as a `docs_new` code owner when convenient? Thanks!

0xtoward committed:
Merge branch 'main' into feature/modernbert-embedding-final
